Feature/input parameters as probability distribution

dhungelgd opened the pull request.

Bachibouzouk committed:
Change variable names and refactor Appliance class

- Mapping between old and new parameters in file constants.py
- Move the class Appliance outside of User class

Bachibouzouk committed:
Create save/export methods for Appliance and User classes

Bachibouzouk committed:
Add UseCase class

A UseCase is simply a list of users

Bachibouzouk committed:
Update core module docstring

Bachibouzouk committed:
Add a function to convert old input files

Bachibouzouk committed:
Improve conversion script

- Renamed Appliance's attribute POWER to power
- Add argument parsing to convert script
- Add lines automatically to the end of .py inputfile
  in order to register the variable names as Appliance instances
  names

Bachibouzouk committed:
Fix loading of specific cycle parameters

Bachibouzouk committed:
Enable running the code from old and new input files

Bachibouzouk committed:
Update README

Bachibouzouk committed:
Fix post-processing and add __init__.py files

Bachibouzouk committed:
Define windows_curve as property of User class

Bachibouzouk committed:
Rename function according to original paper

Move the peak time frame calculation within a function
to allow for unittesting.

Use maximum_profile instead of the misleading window_curve. A User
instance maximum_profile constists of the aggregation of its Appliance
instances maximum_profiles

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move code to get switch on to Appliance class

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move repeated code into Appliance class as a method

Identifies a random switch on time within the available functioning
windows

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move calculation of daily_use update on into a method

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move the rand_total_time_of_use within a method

Bachibouzouk committed:
Use for loop to simplify code readability

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move the code defining random cycles to a method

Also assign the random cycles to an attribute of the appliance

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Remove redundant code in assign_random_cycles method

Also explicitely assign attributes in __init__ method of Appliance class

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Refactor long if statement

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move code to generate appliance profile inside a method

The method is called generate_profile and update the daily_use attribute

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Fix typos in refactoring

Bachibouzouk committed:
Fix docstrings

Bachibouzouk committed:
Add .idea and .pyc to gitignore

Bachibouzouk committed:
Update AUTHORS

Bachibouzouk committed:
Change coincident switch on

Move the code which determines whether the indexes are within or out of
the peak window outside the function, so it makes the argument of the
method more meaningful

Bachibouzouk committed:
Add a parser of xlsx input files

Provide two ways to define power timeseries

Bachibouzouk committed:
Move switch on calibration parameters into a function

Stress the relations between variables and equations of the seminal
paper and the code with comments

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Add a parser of xlsx input files

Provide two ways to define power timeseries

Bachibouzouk committed:
Add options to save converted file

- option -o to provide another output path
- option --suffix to provide a different file name as the python files

Bachibouzouk committed:
Add a test for conversion of input file

This requires pytest to be installed

Bachibouzouk committed:
Add a equality operator to core classes

To compare two user instances and two appliances instances

Bachibouzouk committed:
Reformat example input file

So that the test rewrites them without introducing changes in them

Bachibouzouk committed:
Update requirements

To read the xlsx files

Bachibouzouk committed:
Refactor the random windows calculation

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Refactor the window method of Appliance

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move calculation of coincident switch on into a method

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move code for single user load profile in a method

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Move iteration over user number into a method

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Fix circular import error

Bachibouzouk committed:
Introduce function random_variation

Enhance code readability and decrease risk of errors by typos

Co-authored-by: Gokarna.Dhungel <dhungelgokarna@gmail.com>

Bachibouzouk committed:
Improve docstrings and make variable name PEP8 compatible

Add further quotes to the seminal paper in the docstrings

Bachibouzouk committed:
Change default cycle windows values

dhungelgd committed:
add illustration of giving numerosity input as probability distrbution

dhungelgd committed:
update numerosity to calculate maximum profile

dhungelgd committed:
reset numerosity using the values from input file

Bachibouzouk commented:
We need number in the methods `maximum_profile` and  `calc_coincident_switch_on` of appliance class.

For the maximum it would make sense to get the maximum value of number allowed by the probability distribution, or keep the number if no probability distribution.

For the coincidence switch the appliance number can be given by a method which differentiate whether number was provided as int or as probability distribution

Bachibouzouk committed:
Reformulate definition of appliance number

The user either provides an int or an array of value which is mapped to
0 appliance, 1 appliance, 2 appliances and so on
